homeassistant/components/sunsynk/sensor.py

```diff
@@ -0,0 +1,166 @@
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Solar Energy Total",
+        key=SOLAR_ENERGY_TOTAL,
+        icon="mdi:solar-power",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Grid Import Today",
+        key=GRID_ENERGY_IMPORT_TODAY,
+        icon="mdi:power-plug",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Grid Import Total",
+        key=GRID_ENERGY_IMPORT_TOTAL,
+        icon="mdi:power-plug",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Grid Export Today",
+        key=GRID_ENERGY_EXPORT_TODAY,
+        icon="mdi:power-plug",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Grid Export Total",
+        key=GRID_ENERGY_EXPORT_TOTAL,
+        icon="mdi:power-plug",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a Sunsynk entry."""
+    inverter_sn = entry.data[DATA_INVERTER_SN]
+
+    coordinator: SunsynkCoordinator = hass.data[DOMAIN][entry.entry_id][
+        SUNSYNK_COORDINATOR
+    ]
+
+    sensors = [
+        SunsynkSensor(coordinator, description, inverter_sn)
+        for description in SENSOR_TYPES
+    ]
+
+    async_add_entities(sensors)
+
+
+class SunsynkSensor(CoordinatorEntity[SunsynkCoordinator], SensorEntity):
+    """Representation of a grid power usage."""
+
+    def __init__(
+        self,
+        coordinator: SunsynkCoordinator,
+        description: SensorEntityDescription,
+        inverter_sn: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self.inverter_sn = inverter_sn
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.inverter_sn)},
+            name=f"Inverter {self.inverter_sn}",
+            manufacturer="Sunsynk",
+        )
+
+    @property
+    def unique_id(self) -> str | None:
```

Comment:
Can be a shorthand attribute

Reply:
Sorry, I'm not quite sure what you mean? Please can elaborate or provide an example?

Reply:
Oh right, you can change this into `_attr_unique_id = ...` and it will work the same

Reply:
Thank you, I've updated this now and re-tested